datajoint/external.py

```diff
@@ -338,7 +338,7 @@ def delete(self, *, delete_external_files=None, limit=None, display_progress=Tru
             error_list = []
             for uuid, external_path in items:
```

Comment:
Thinking it might be safer to perform a `LOCK` on the external table before starting the loop to ensure that no references are created while the remote file is being deleted. Just have to be sure that we `UNLOCK` so that we don't cause issues for others. This will reinforce that this is an admin operation and should be done carefully.

https://dev.mysql.com/doc/refman/5.7/en/lock-tables.html